### PR TITLE
fix: Numbered Lists not incremental

### PR DESCRIPTION
**Pull request by buddy-web3**

### CODE REVIEW

File: parser/parser.js [hunk 1]
```diff
@@ -249,12 +250,9 @@ export const parseParagraph = (documentContext) => (paragraph) => {
...
       currentLevel.glyphType !== "GLYPH_TYPE_UNSPECIFIED";
 
-    // Please forgive me for always using 1. as the sequence number on list items
-    // It's sorta hard to count them properly so I'm depending on markdown renderers doing the heavy lifting for me.
-    // Which, in fairness, they're supposed to.
-    itemMarker = isOrdered ? "1. " : "- ";
+    itemMarker = isOrdered ? orderedList + ". " || "1. " : "- ";
     leadingSpace = new Array(nestingLevel).fill("    ").join("");
-
+    documentContext.orderedList++;
```
Comment (mruwnik) [inline parser/parser.js:255]: does this mean that there can only be one ordererd list? What if someone wants a couple in the same doc?

File: parser/__tests__/parser.test.js [hunk 2]
```diff
@@ -356,6 +356,7 @@ describe("parseParagraph", () => {
       ...paragraph,
       bullet: { nestingLevel: 1, listId: "list-id" },
     };
+    documentContext.orderedList = 1;
```
Comment (mruwnik) [inline parser/__tests__/parser.test.js:359]: can you add some more test cases for:
* No orderedList set (should use the default)
* More than one item (so the number goes up)
* Multiple numbered lists in the same document
* Multiple lists, both numbered and unnumbered in the same document

**Comment by buddy-web3** [comment]
@ChrisRimmer 
`The markdown spec dictates that numbered lists are allowed to all be prefixed with the same number, and it's up to the renderer to behave correctly so the existing behaviour is entirely supported.`
 - markdown does not behave as described when there are new lines between the list items.
 https://github.com/StampyAI/stampy-ui/issues/352

File: parser/parser.js (hunk 1 above)
Comment (buddy-web3) [reply parser/parser.js:255]: point taken, will update the code.

File: parser/__tests__/parser.test.js (hunk 2 above)
Comment (buddy-web3) [reply parser/__tests__/parser.test.js:359]: noted

**Comment by ChrisRimmer** [comment]
> @ChrisRimmer 
> 
> markdown does not behave as described when there are new lines between the list items.

Fair enough. Erroneous gaps between list items seems like the more important thing to fix but who am I to dictate, I haven't worked on this in almost a year 😅

File: parser/parser.js (hunk 1 above)
Comment (buddy-web3) [reply parser/parser.js]: this has been resolved.

File: parser/__tests__/parser.test.js (hunk 2 above)
Comment (buddy-web3) [reply parser/__tests__/parser.test.js]: I'm having troubles getting the testsuite to write more than 1  line. adding `\n` makes the second line be part of the same Item, on a new line, padded.
Could you please give me an example test which has more than 1 item?

Comment (mruwnik) [reply parser/__tests__/parser.test.js]: try using whatever gdocs returns for the doc that is causing this issue (after changing it to not be so verbose, of course)

Comment (jrhender) [reply parser/__tests__/parser.test.js]: Hey @buddy-web3 , are you still working on this PR?
As far as I can tell, @mruwnik 's requested test cases haven't yet been added. Is this right? If so, happy to try to add the requested cases.

Comment (mruwnik) [reply parser/__tests__/parser.test.js]: I reckon you can take this over

**Comment by LeMurphant** [comment]
Rendered obsolete by https://github.com/StampyAI/GDocsRelatedThings/pull/61

**Pull request closed without merging**